File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystProfileTag.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * A profile tag
 */
@Serializable
data class CatalystProfileTag(
    val id: String,
    val name: String,
    val normalizedName: String
)

/**
 * Wrapper for the `{"tags": [...]}` response shape
 */
@Serializable
internal data class CatalystProfileTagsWrapper(val tags: List<CatalystProfileTag>)

/**
 * A profile tag suggestion, including usage count
 */
@Serializable
data class CatalystProfileTagSuggestion(
    val id: String,
    val name: String,
    val usageCount: Int
)

/**
 * Wrapper for the `{"tags": [...]}` response shape (profile tag suggestions)
 */
@Serializable
internal data class CatalystProfileTagSuggestionsWrapper(val tags: List<CatalystProfileTagSuggestion>)

/**
 * A user matched by a profile tag search, including which tags matched
 */
@Serializable
data class CatalystProfileTagUser(
    val id: String,
    val screenName: String,
    val displayName: String,
    val profile: EgeriaUserProfile? = null,
    val profileEmoji: EgeriaUserProfileEmoji? = null,
    val matchedTags: List<String>
)

/**
 * Summary information for a profile tag (used within [CatalystProfileTagUsersResult])
 */
@Serializable
data class CatalystProfileTagSummary(
    val name: String,
    val usageCount: Int
)

/**
 * Result of listing users by profile tag, including cursor-based pagination
 */
@Serializable
data class CatalystProfileTagUsersResult(
    val tag: CatalystProfileTagSummary,
    val users: List<CatalystProfileTagUser>,
    val nextCursor: String? = null
)

File: packages/swift/Sources/CatalystSwift/Models/CatalystProfileTag.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// A profile tag.
public struct CatalystProfileTag: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let name: String
  public let normalizedName: String
}

/// Wrapper for the `{"tags": [...]}` response shape.
public struct CatalystProfileTagsWrapper: Decodable, Sendable {
  public let tags: [CatalystProfileTag]
}

/// A profile tag suggestion, including usage count.
public struct CatalystProfileTagSuggestion: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let name: String
  public let usageCount: Int
}

/// Wrapper for the `{"tags": [...]}` response shape (profile tag suggestions).
public struct CatalystProfileTagSuggestionsWrapper: Decodable, Sendable {
  public let tags: [CatalystProfileTagSuggestion]
}

/// A user matched by a profile tag search, including which tags matched.
public struct CatalystProfileTagUser: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let screenName: String
  public let displayName: String
  public let profile: EgeriaUserProfile?
  public let profileEmoji: EgeriaUserProfileEmoji?
  public let matchedTags: [String]
}

/// Summary information for a profile tag (used within `CatalystProfileTagUsersResult`).
public struct CatalystProfileTagSummary: Decodable, Sendable, Equatable, Hashable {
  public let name: String
  public let usageCount: Int
}

/// Result of listing users by profile tag, including cursor-based pagination.
public struct CatalystProfileTagUsersResult: Decodable, Sendable {
  public let tag: CatalystProfileTagSummary
  public let users: [CatalystProfileTagUser]
  public let nextCursor: String?
}
